src/overlays/actors/ovl_Bg_Hidan_Hrock/z_bg_hidan_hrock.c

```diff
@@ -65,21 +69,174 @@ static ColliderTrisInit D_80889898 = {
         COLSHAPE_TRIS,
     },
     2,
-    D_80889820,
+    sTrisElementsInit,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(gravity, -1, ICHAIN_STOP),
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Hrock/BgHidanHrock_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Hrock/BgHidanHrock_Destroy.s")
+void BgHidanHrock_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanHrock* this = THIS;
+    ColliderTrisElementInit* colliderElementInit;
+    Vec3f vertices[3];
+    f32 cosRotY;
+    f32 sinRotY;
+    s32 i;
+    s32 j;
+    CollisionHeader* collisionHeader;
+
+    collisionHeader = NULL;
+    Actor_ProcessInitChain(thisx, sInitChain);
+    this->unk_16A = thisx->params & 0x3F;
+    thisx->params = (thisx->params >> 8) & 0xFF;
+    Collider_InitTris(globalCtx, &this->collider);
+    Collider_SetTris(globalCtx, &this->collider, thisx, &sTrisInit, this->colliderItems);
+    DynaPolyActor_Init(&this->dyna, DPM_UNK);
+
+    sinRotY = Math_SinS(thisx->shape.rot.y);
+    cosRotY = Math_CosS(thisx->shape.rot.y);
+
+    if (thisx->params == 0) {
+        sinRotY *= 1.5f;
+        cosRotY *= 1.5f;
+    }
+
+    for (i = 0; i < 2; i++) {
+        colliderElementInit = &sTrisInit.elements[i];
+
+        if (1) {
+            for (j = 0; j < 3; j++) {
+                Vec3f* vtx = &colliderElementInit->dim.vtx[j];
+
+                vertices[j].x = vtx->z * sinRotY + (thisx->home.pos.x + vtx->x * cosRotY);
+                vertices[j].y = vtx->y + thisx->home.pos.y;
+                vertices[j].z = vtx->z * cosRotY + (thisx->home.pos.z - vtx->x * sinRotY);
+            }
+        }
+        Collider_SetTrisVertices(&this->collider, i, &vertices[0], &vertices[1], &vertices[2]);
+    }
```

Comment:
This double for loop usually works fine with `sTrisInit.elements[i].dim.vtx[j]` instead of `vtx`. Did you try that?

Reply:
I just tried that and it unfortunately didn't work.. this is a little more readable anyway.